From 9806c6380d9f0828bbaab3978b9393541e090b1d Mon Sep 17 00:00:00 2001
From: Dean Sanner <dsanner@us.ibm.com>
Date: Sat, 21 Oct 2017 22:05:10 -0500
Subject: [PATCH] Fix PNOR sections loaded into reserve-mem for non secure sys

 - The sha512 header was getting double removed.  Add it back
   into the reserved mem section in non secureboot configs

Change-Id: I3ca4afe5851ee239b4d41e5e8f1871004103161f
---
 src/usr/runtime/populate_hbruntime.C | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/usr/runtime/populate_hbruntime.C b/src/usr/runtime/populate_hbruntime.C
index dbf35db..b1a2781 100644
--- a/src/usr/runtime/populate_hbruntime.C
+++ b/src/usr/runtime/populate_hbruntime.C
@@ -639,6 +639,13 @@ errlHndl_t hbResvLoadSecureSection (const PNOR::SectionId i_sec,
             l_pnorVaddr -= PAGESIZE;
             l_imgSize += PAGESIZE;
         }
+#else
+         if (l_info.sha512Version)
+        {
+            // Include sha512Version header
+            l_pnorVaddr -= PAGESIZE;
+            l_imgSize += PAGESIZE;
+        }
 #endif
 
         // Align size for OPAL
-- 
1.8.2.2

